core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -1634,6 +1634,17 @@ default Optional<TopNApplicationResult<ConnectorTableHandle>> applyTopN(
         return Optional.empty();
     }
 
+    /**
+     * If the file is ordered, attempt to push down sort or topN to connector.
+     */
+    default Optional<PartialSortApplicationResult<ConnectorTableHandle>> applyPartialSort(
```

Comment:
In Trino partial sort would often indicate "distributed sort" feature, but this is not the context this method is called (or could be called in the future). Not sure how this new SPI method should be best framed.

cc @martint